crypto/bio/b_addr.c

```diff
@@ -760,8 +762,11 @@ int BIO_lookup_ex(const char *host, const char *service, int lookup_type,
                 he_fallback_address = INADDR_ANY;
                 break;
             default:
-                OPENSSL_assert(("We forgot to handle a lookup type!" == 0));
-                break;
+                /* We forgot to handle a lookup type! */
+                assert("We forgot to handle a lookup type!" == NULL);
```

Comment:
Because we've already detected an internal error condition if we get here so using ossl_assert() doesn't really add anything. I actually originally had assert(0) here but in the PR that preceded this one @appro asked me to put it back to this. I'm not sure we need a new macro for this. It's infrequent enough not to need it.